HAWQ-915. RAT is failing on the latest master

paul-guo- opened the pull request.

changleicn commented:
@rvs to review.

rvs commented on pom.xml:
```diff
@@ -37,14 +37,32 @@
               <!-- stream benchmark under BSD-like license with a few exceptions around benchmarking results -->
               <exclude>tools/bin/src/stream/*</exclude> 
 
-              <!-- PyGreSQL an open-source Python module that interfaces to a PostgreSQL database under the Python license -->
```
Please leave this comment near PyGreSQL entry

rvs commented on pom.xml:
```diff
@@ -37,14 +37,32 @@
+              <!-- exclude open-source python modules -->
+              <!-- MIT license for open-source python modules -->
               <exclude>tools/bin/pythonSrc/PyGreSQL-4.0/**</exclude>
+              <exclude>tools/bin/pythonSrc/PSI-0.3b2_gp/**</exclude>
```
For all these new additions don't forget to update the NOTICE file

rvs commented on pom.xml:
```diff
@@ -37,14 +37,32 @@
+              <exclude>tools/bin/pythonSrc/lockfile-0.9.1/**</exclude>
+              <!-- BSD license for open-source python modules -->
+              <exclude>tools/bin/pythonSrc/pychecker-0.8.18/**</exclude>
+              <exclude>tools/bin/pythonSrc/unittest2-0.5.1/**</exclude>
```
unittest2-0.5.1/PKG-INFO says License is unknown. Please provide support for your claim that this is a BSD licensed code.

paul-guo- replied:
PKG-INFO is wrong (just noticed).

tools/bin/pythonSrc/unittest2-0.5.1]$ head setup.py

#! /usr/bin/env python
# setup.py
# Install script for unittest2
# Copyright (C) 2010 Michael Foord
# E-mail: fuzzyman AT voidspace DOT org DOT uk

# This software is licensed under the terms of the BSD license.
# http://www.voidspace.org.uk/python/license.shtml

import os

2016-07-13 7:58 GMT+08:00 Roman V Shaposhnik <notifications@github.com>:

> In pom.xml
> <https://github.com/apache/incubator-hawq/pull/788#discussion_r70544327>:
>
> >                <exclude>tools/bin/pythonSrc/PyGreSQL-4.0/**</exclude>
> > +              <exclude>tools/bin/pythonSrc/PSI-0.3b2_gp/**</exclude>
> > +              <exclude>tools/bin/pythonSrc/lockfile-0.9.1/**</exclude>
> > +              <!-- BSD license for open-source python modules -->
> > +              <exclude>tools/bin/pythonSrc/pychecker-0.8.18/**</exclude>
> > +              <exclude>tools/bin/pythonSrc/unittest2-0.5.1/**</exclude>
>
> unittest2-0.5.1/PKG-INFO says License is unknown. Please provide support
> for your claim that this is a BSD licensed code.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-hawq/pull/788/files/0c825258cc886917477b60f706a87e7d3dd831f4#r70544327>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHI5jGA59Fdr2gVGgdY5tZOVHFkHDYy_ks5qVCo7gaJpZM4JKQQE>
> .
>

rvs commented on pom.xml:
```diff
@@ -37,14 +37,32 @@
+              <exclude>tools/bin/pythonSrc/pychecker-0.8.18/**</exclude>
+              <exclude>tools/bin/pythonSrc/unittest2-0.5.1/**</exclude>
+              <!-- apache compatible license for open-source python modules -->
+              <exclude>tools/bin/pythonSrc/pycrypto-2.0.1/**</exclude>
```
At this point the agreement is that we're not distributing crypto code with HAWQ

rvs commented on pom.xml:
```diff
@@ -37,14 +37,32 @@
+              <!-- BSD license for open-source python modules -->
+              <exclude>tools/bin/pythonSrc/pychecker-0.8.18/**</exclude>
+              <exclude>tools/bin/pythonSrc/unittest2-0.5.1/**</exclude>
+              <!-- apache compatible license for open-source python modules -->
```
I am not sure this is ASF compatible since this code basically has no OSI-approved license attached to it

rvs commented on pom.xml:
```diff
@@ -37,14 +37,32 @@
+              <!-- apache compatible license for open-source python modules -->
+              <exclude>tools/bin/pythonSrc/pycrypto-2.0.1/**</exclude>
+              <!-- Apache compatible open-source python modules under other directories -->
+              <exclude>tools/bin/ext/**</exclude>
```
I am no comfortable with bulk exclusion here, especially since the code is Crypto related in one of the folders

rvs commented on pom.xml:
```diff
@@ -37,14 +37,32 @@
+              <exclude>tools/bin/ext/**</exclude>
+
+              <!-- Apache license for open-source thrift  -->
+              <exclude>depends/thirdparty/thrift/**</exclude>
```
What exactly is RAT complaining about? For ASF code this should not be a problem.

rvs commented on pom.xml:
```diff
@@ -37,14 +37,32 @@
               <exclude>depends/libyarn/CMake/FindBoost.cmake</exclude>
               <exclude>depends/libyarn/CMake/FindGSasl.cmake</exclude>
               <exclude>depends/libyarn/CMake/FindKERBEROS.cmake</exclude>
+			  <exclude>depends/libyarn/CMake/FindGoogleTest.cmake</exclude>
```
Where did this file come from? If you wrote it, please add AL header to it. If not -- indicate where it came from.

rvs commented on pom.xml:
```diff
@@ -37,14 +37,32 @@
               <exclude>depends/libyarn/CMake/FindKERBEROS.cmake</exclude>
+			  <exclude>depends/libyarn/CMake/FindGoogleTest.cmake</exclude>
 			  <exclude>depends/libhdfs3/CMake/FindBoost.cmake</exclude>
+			  <exclude>depends/libhdfs3/CMake/FindGoogleTest.cmake</exclude>
```
Where did this file come from? If you wrote it, please add AL header to it. If not -- indicate where it came from.

rvs commented on pom.xml:
```diff
@@ -112,6 +130,10 @@
 
+              <!-- Files which are not easy to have license headers. -->
+              <exclude>depends/libhdfs3/test/data/*</exclude>
+              <exclude> depends/libhdfs3/debian/**</exclude>
```
a small nit: please fix a spurious whitespace

paul-guo- commented:
Thanks @rvs , I've made a lot of further changes.

Removed some deleted files in pom.xml
Added header info into more files after discussing with related developers.
Classified and double-confirmed licenses for opensource python modules under tools/bin.

Just one thing, pygresql is said to be Python Software Foundation License (https://pypi.python.org/pypi/PyGreSQL/4.0) (Note Python license is deprecated one according to https://en.wikipedia.org/wiki/Python_Software_Foundation_License, so Python license != Python Software Foundation License) but the code & doc (docs/readme.txt & pgmodule.c) shows it is with postgresql license. I'm really confused. Fortunately it seems that both licenses are friendly to ASF license.

I'm classifying pygresql as Python Software Foundation License in this patch, since this is not a big issue. We could keep the issue open and make the change if necessary.

paul-guo- commented:
@radarwave Could you please take a look at the patch also?

rvs commented:
@paul-guo- great sleuthing! The patch looks good to me now. Can you please commit?

radarwave commented:
+1
